CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
It's 3.0 for GTSAM, I think we need to discuss on this.

Reply:
Yeah we need to upgrade GTSAM's cmake to 3.12 as well. The old FindPython API was deprecated in 3.12 that's why I went with the new one.

Reply:
Actually the 2 versions should not matter, since GTSAM on its own works with 3.0. If a user needs to use wrap, it should be straightforward to use an upgraded CMake version.

Reply:
@ProfFan ping. Would you like to discuss this and get it over the hill?

Reply:
I would say I am totally fine with this, just Frank may have different views :)

Reply:
@dellaert comments? I don't want to have the headache of maintaining CMake for multiple versions when users can simply use 3.12 (released 11/2018) or higher. The current latest version is 3.19.